packages/arb-token-bridge-ui/src/components/TransferPanel/TokenApprovalDialog.tsx

```diff
@@ -91,11 +89,14 @@ export function TokenApprovalDialog(props: TokenApprovalDialogProps) {
         if (!signer) {
           gasEstimate = constants.Zero
         } else {
-          gasEstimate = await approveCctpEstimateGas(
-            chainId,
-            constants.MaxUint256,
+          const cctpTransferStarter = new CctpTransferStarter({
+            sourceChainProvider: provider,
+            destinationChainProvider: isDepositMode ? l2.provider : l1.provider
```

Comment:
this way it won't be tied to what chain the user is currently connected to, and plays nicely with https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1307

```suggestion
            sourceChainProvider: isDepositMode ? l1.provider : l2.provider,
            destinationChainProvider: isDepositMode ? l2.provider : l1.provider
```

Reply:
I've made the change, it looks good from a consistency pov, but the `provider` variable that was being used here is defined originally as `const provider = isDepositMode ? l1.provider : l2.provider` only.